src/parser/mod.rs

```diff
@@ -11384,6 +11406,23 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parses MSSQL's json-null-clause
+    fn parse_json_null_clause(&mut self) -> Option<JsonNullClause> {
+        if !dialect_of!(self is MsSqlDialect) {
+            return None;
+        }
+        if let Some(kw) = self.parse_one_of_keywords(&[Keyword::ABSENT, Keyword::NULL]) {
+            if self.parse_keywords(&[Keyword::ON, Keyword::NULL]) {
+                if kw == Keyword::ABSENT {
+                    return Some(JsonNullClause::AbsentOnNull);
+                } else if kw == Keyword::NULL {
+                    return Some(JsonNullClause::NullOnNull);
+                }
+            }
+        }
```

Comment:
could we go back to the previous version of this logic? I think that was more readable since we avoided conditional state, and it read the same as the syntax. (e.g its no longer super clear what's happening if we match `ABSENT` but not the `ON NULL` part)
```rust
if self.parse_keywords(absent, on, null) {
   Some()
else if self.parse_keywords(null, on, null) {
   Some()
} else {
    None
}
```